### ISSUE TITLE
MAC Vendor Lookup

### ISSUE THREAD

**Issue by gd1eh** [cloudtracer/ThreatPinchLookup#34]
Hi, 

I have created MAC Vendor lookup functionality which seems to have broke since the last update. You can add it to Threatpinch as it can be useful (when it works).

MAC Lookup type:

"MAC": {
    "disabled": false,
    "height": 217,
    "hoverdelay": 2000,
    "name": "MAC",
    "popuplength": 720000,
    "regex": "(([0-9A-Za-z]{2}(-)){5}[0-9A-Za-z]{2}|([0-9A-Za-z]{2}(\\:)){5}[0-9A-Za-z]{2}|([0-9A-Za-z]{2}(\\.)){5}[0-9A-Za-z]{2})",
    "width": 280
  }

MAC Request Lookup:

"MACVENDORLOOKUP": {
    "lookupName": "MAC Vendor Lookup",
    "userDefined": false,
    "lookupVariable": "MACVENDORLOOKUP",
    "lookupType": "MAC",
    "lookupUrl": "http://macvendors.co/api/${PINCH.HOVERITEM}",
   "httpHeaders": "",
    "httpType": "GET",
    "dataType": "JSON",
    "dataSchema": {
      "title": {
        "title": "Macvendors.co",
        "mapping": "${PINCH.LINKURL}",
        "order": 0,
        "linkTitle": "${PINCH.HOVERITEM}",
        "linkUrl": "macvendors.co/results/${PINCH.HOVERITEM}"
      },
      "company": {
        "title": "company",
        "mapping": "${PINCH.RESPONSE.result.company}",
        "order": 1
      },
      "country": {
        "title": "country",
        "mapping": "${PINCH.RESPONSE.result.country}",
        "order": 2
      }
    },
    "authorizationType": "DEFAULT",
    "order": 34,
    "requestGroup": "INTERNET"
  }

**Comment by cloudtracer** [comment]
Very cool! Thats awesome gd1eh! I will include these in a future release!  Regarding the MAC Lookup, I've tweaked the regex a bit and got it working on my own plugin now. It looks like some of the escape slashes are missing from yours. Its possible the regex string might have been unescaped during the update process, I'll look into it.

  "MAC": {
    "disabled": false,
    "height": 217,
    "hoverdelay": 2000,
    "name": "MAC",
    "popuplength": 720000,
    "regex": "(\\b(([0-9A-Za-z]{2}(-)){5}[0-9A-Za-z]{2}|([0-9A-Za-z]{2}(\\:)){5}[0-9A-Za-z]{2}|([0-9A-Za-z]{2}(\\.)){5}[0-9A-Za-z]{2})\\b)",
    "width": 280
  },

Looks like you started using the Wizard early! Very cool, but there was an issue with the earlier version of it where it would set the userDefined attribute to false if there weren't any user defined variables and this can cause the plugin to become unstable.

Walk through this wizard url and it should fix the userDefined false issue for the request lookup: chrome-extension://fkhklenliphocojipngadahdilpnhdfh/src/options/wizard.html?RL=eyJsb29rdXBOYW1lIjoiTUFDIFZlbmRvciBMb29rdXAiLCJsb29rdXBWYXJpYWJsZSI6Ik1BQ1ZFTkRPUkxPT0tVUCIsImxvb2t1cFR5cGUiOiJNQUMiLCJsb29rdXBVcmwiOiJodHRwOi8vbWFjdmVuZG9ycy5jby9hcGkvJHtQSU5DSC5IT1ZFUklURU19IiwiaHR0cEhlYWRlcnMiOiIiLCJodHRwVHlwZSI6IkdFVCIsImRhdGFUeXBlIjoianNvbiIsImRhdGFTY2hlbWEiOnsiMF9NYWN2ZW5kb3JzLmNvIjp7InRpdGxlIjoiTWFjdmVuZG9ycy5jbyIsIm1hcHBpbmciOiIke1BJTkNILkxJTktVUkx9Iiwib3JkZXIiOjAsImxpbmtUaXRsZSI6IiR7UElOQ0guSE9WRVJJVEVNfSIsImxpbmtVcmwiOiJtYWN2ZW5kb3JzLmNvL3Jlc3VsdHMvJHtQSU5DSC5IT1ZFUklURU19In0sIjFfY29tcGFueSI6eyJ0aXRsZSI6ImNvbXBhbnkiLCJtYXBwaW5nIjoiJHtQSU5DSC5SRVNQT05TRS5yZXN1bHQuY29tcGFueX0iLCJvcmRlciI6MX0sIjJfY291bnRyeSI6eyJ0aXRsZSI6ImNvdW50cnkiLCJtYXBwaW5nIjoiJHtQSU5DSC5SRVNQT05TRS5yZXN1bHQuY291bnRyeX0iLCJvcmRlciI6Mn19LCJkaXNhYmxlZCI6ZmFsc2UsImlvY3MiOmZhbHNlLCJhdXRob3JpemF0aW9uVHlwZSI6IkRFRkFVTFQiLCJyZXF1ZXN0R3JvdXAiOiJJTlRFUk5FVCIsInNhbXBsZSI6IjA4OjAwOjY5OjAyOjAxOkZDIiwib3JkZXIiOjM5fQ==

**Comment by gd1eh** [comment]
Thanks, it seems that the update did change that indeed. Got it working again. 
Glad to help. Will share if I come up with new things. Have some fun ideas though but would rather have another communication platform to go over those. 

Well, I remember that I ran into some issues when using the userDefined to true before. It would literally break the "settings" and other functions. I manually set this to false to fix that.
